### ISSUE TITLE
Fix github deprecation of set-output

### ISSUE THREAD

**Issue by skeeping** [agilepathway/label-checker#316]
Hello i have created a PR with the fix (i believe it should work) https://github.com/agilepathway/label-checker/pull/315 

i followed the instructions on https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ and it has worked on my other projects.

**Comment by johnboyes** [comment]
Thanks very much for your contribution, @skeeping 🙏🏻 
Very happy to accept this.

I see that some of the required automated checks are failing on #315, grateful if you could take a look and see if you can fix them, please?  See the [contributing guide](https://github.com/agilepathway/label-checker/blob/master/CONTRIBUTING.md) for more info, and happy to help if you're unclear about what needs to be changed to get all the automated checks to pass 👍🏻

**Comment by johnboyes** [comment]
It's getting closer, just 2 things to be fixed now:

1. the [tests](https://github.com/agilepathway/label-checker/actions/runs/3376739888/jobs/5604861262)
2. the commit messages need to satisfy the [opinionated git commit message checker](https://github.com/mristin/opinionated-commit-message), i.e. in [this style](https://chris.beams.io/posts/git-commit/). I'd recommend rebasing the commits into just one commit with a commit message in that style on your branch and force pushing, if that's ok?   I have the checker as I really value having the commit history in a consistent style.

**Comment by skeeping** [comment]
1. of course. 
2. definitely can fix. How i run this on my repos is by doing squash and merge as the strategy and using the PR as the commit title/message but i will update it and squash it all together once i have sort out the tests :)

**Comment by johnboyes** [comment]
Great stuff, thanks.  I know what you mean about squashing when merging the PR, I guess I'm a zealot as I like to have the individual commits inside the PR being in the same style too - makes it easier when looking back on PRs using the GitHub UI.  Could be seen as overkill I know 😉

**Comment by skeeping** [comment]
No worries better to be overkill then lost later on down the line :) 

That being said i have no idea how to fix the test unfortunately :(

**Comment by johnboyes** [comment]
All good, I'll look into the tests and get back to you as soon as I have time (hopefully this evening, if not then at some point in the next few days)

**Comment by skeeping** [comment]
Thanks i will fix the commit message in the mean time

**Comment by johnboyes** [comment]
The commit message is very nearly compliant with the opinionated commit message checker now, but the subject line is (just) [too long](https://github.com/agilepathway/label-checker/actions/runs/3376961723/jobs/5617148103). 
Changing the subject to something like this will fix it:

> Remove deprecated \`set-output\` in GitHub Actions

**Comment by skeeping** [comment]
sorry i removed a couple characters because i thought it was only 1 to long but that must not have been the case :(

**Comment by skeeping** [comment]
have you had any chance to look at this?

**Comment by johnboyes** [comment]
> have you had any chance to look at this?

Sorry for delay.  I've looked into it and added comments to the PR with the amendments required.  I've verified that these amendments fix the tests, so after you have made them this PR will be good to go. 👍🏻

**Comment by skeeping** [comment]
thanks for the help and its done now :)

**Comment by johnboyes** [comment]
Ah, I meant to add the `\n` at the end of the `echo` statement like this, for example:

`"echo 'label_check=failure' >> $GITHUB_OUTPUT\n"`

The way you have done it the tests still fail.

Hold off on amending it again for now.  I've realised that I need to fully integration test this.  I'll do that today and report back.

**Comment by skeeping** [comment]
sorry didnt read the whole message but it has been changed to the end of the echo statement. Let me know if there is anything else

**Comment by johnboyes** [comment]
I integration tested it and all worked fine, so went ahead and released this: [release v1.2.8](https://github.com/agilepathway/label-checker/releases/tag/v1.2.8)

Thanks again for your contribution @skeeping 👍🏻

**Comment by johnboyes** [comment]
v1.2.8 has now been reverted via [v1.2.9](https://github.com/agilepathway/label-checker/releases/tag/v1.2.9), which means that this issue #316 is unsolved for now.  I'll only attempt another fix for it in a month or two at least, to ensure stability of the label checker now.

**Comment by johnboyes** [comment]
Hi @skeeping, I've released [v1.2.13](https://github.com/agilepathway/label-checker/releases/tag/v1.2.13) which replaces the deprecated `set-output`.  The reason why the code in your original PR didn't work is that because the label checker is a [Docker container action](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action).  That means that using `echo` to append to the `GITHUB_OUTPUT` environment variable doesn't work, and instead we need to write to the `GITHUB_OUTPUT` environment file directly.

Your contribution was very useful for me though, as it made me realise that while the label checker integration tests give good coverage, they do not cover everything.  So before making this change I added some [end-to-end tests](https://github.com/agilepathway/label-checker/blob/master/.github/workflows/end_to_end_test.yml) which ensured I had this functionality properly covered (and actually give better test coverage in a much more elegant way than the integration tests).  So thanks for your contribution as it has helped me to improve!
